plugins/woocommerce/src/Internal/ProductAttributesLookup/LookupDataStore.php

```diff
@@ -648,4 +657,28 @@ public function set_regeneration_in_progress_flag() {
 	public function unset_regeneration_in_progress_flag() {
 		delete_option( 'woocommerce_attribute_lookup_regeneration_in_progress' );
 	}
+
+	/**
+	 * Set a flag indicating that the last lookup table regeneration process started was aborted.
+	 */
+	public function set_regeneration_aborted_flag() {
+		update_option( 'woocommerce_attribute_lookup_regeneration_aborted', 'yes' );
+	}
+
+	/**
+	 * Remove the flag indicating that the last lookup table regeneration process started was aborted.
+	 */
+	public function unset_regeneration_aborted_flag() {
+		delete_option( 'woocommerce_attribute_lookup_regeneration_aborted' );
+	}
+
+	/**
+	 * Tells if the last lookup table regeneration process started was aborted
+	 * (via deleting the 'woocommerce_attribute_lookup_regeneration_in_progress' option).
+	 *
+	 * @return bool True if the last lookup table regeneration process was aborted.
+	 */
+	public function regeneration_was_aborted() {
```

Comment:
Minor thing, but we could use return types here:

```suggestion
	public function regeneration_was_aborted(): bool {
```